### DISCUSSION TITLE
Computed fields with extra="forbid"

### DISCUSSION THREAD

**Discussion by stur86** [pydantic/pydantic#7782] [Question]
I'm finding a bit of a contradictory behaviour. I create a model with a computed field, e.g.:

```python
class MyModel(BaseModel, extra="forbid"):
    x: float = 0.0
    
    @computed_field
    @property
    def x_squared(self) -> float:
        return x**2
```

But then parsing back the JSON produced by this same model results in an error, because `x_squared` gets written to JSON but counts as an extra field. Now I *do* want it to be written out, I just don't want it to cause an error when re-reading. Is there any way to tag this value, specifically, for ignoring when validating? Or do I have to go the long way with a model validator?

**Comment by billou57** [comment]
Hello. You could write a custom `__init__` in which you delete the field before using `super()`

**Reply by stur86** [reply]
I tried doing something similar but with a validator; is the `__init__` guaranteed to be called even when using `model_validate` then?

**Comment by schoelst** [comment]
Hi!
I am using a `model_validator` to remove the computed fields

```python

    @model_validator(mode="before")
    def remove_properties(cls, values: dict) -> dict:
        for p in ["x_squared"]:
            if p in values:
                del values[p]
        return values
```

It's not beautiful but it works. I would have expected that properties that have the `@computed_field` decorator get removed from the passed values automatically such that serialization followed by deserialization works.
